skimage/filters/_gaussian.py

```
@@ -112,9 +112,17 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
             sigma = [sigma] * (image.ndim - 1)
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
-    image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output is not None:
+        if isinstance(output, np.ndarray):
+            # convert image to float if output is float numpy ndarray
+            if issubclass(output.dtype.type, np.floating):
+                image = convert_to_float(image, preserve_range)
+        elif issubclass(type(output), type):
+            # convert image to float if output is float dtype
+            if issubclass(output, (np.floating, float)):
+                image = convert_to_float(image, preserve_range)
```

I again, personally, prefer to raise an error if output is not an array.